Shoko.Server/Renamer/RenameFileHelper.cs

```diff
@@ -172,11 +180,17 @@ internal static void FindRenamers(IList<Assembly> assemblies)
                     if (Renamers.ContainsKey(key))
                     {
                         Logger.Warn(
-                            $"[RENAMER] Warning Duplicate renamer key \"{key}\" of types {implementation}@{implementation.Assembly.Location} and {Renamers[key]}@{Renamers[key].type.Assembly.Location}");
+                            $"[RENAMER] Warning Duplicate renamer key \"{key}\" of types {implementation}@{implementation.Assembly.Location} and {Renamers[key]}@{Renamers[key].Type.Assembly.Location}");
                         continue;
                     }
 
-                    Renamers.Add(key, (implementation, desc));
+                    Renamers.Add(key, new RenamerInfo
+                    {
+                        ID = Renamers.Count + 1,
+                        Name = key,
```

Comment:
You're double storing key here, which is pointless.

Renamer ID should never be dynamically generated as it will be confusing to runtime, what is even the reason, you mentioned in discord "Comprimises" but I can't think of anything at this point in time.

Reply:
Just tested it and asp.net can handle dynamic routes with strings and static routes just fine.